### ISSUE TITLE
Better structure for labels

### ISSUE THREAD

**Issue by davidmehren** [hedgedoc/hedgedoc#589]
In an effort to make contributing to HedgeDoc easier, I'd like to introduce a common structure and set of labels for issues and PRs across the HedgeDoc org.

I came up with these labels as a baseline:

## Type
What is the issue/PR about?

- `type: bug`
    - Something isn't working
- `type: discussion`
    - Something needs to be planned/discussed
- `type: documentation`
    - Things relating to our docs
- `type: enhancement`
    - An improvement to existing functionality
- `type: feature`
    - Adds or requests new functionality
- `type: question`
    - Somebody has a question
- `type: maintenance`
    - Regular maintenance

## Priority
How important is it?

- `priority: critical`
    - Something that blocks a release
- `priority: high`
    - Something needed by most users
- `priority: medium`
    - Something needed by some users
- `priority: low`
    - Something needed by few users

## Status
What is the status? Can/will this be worked on?

- `status: can't reproduce`
    - We cannot reproduce the problem
- `status: confirmed`
    - We have confirmed that the issue exists
- `status: duplicate`
    - This issue or pull request already exists
- `status: needs information`
    - Some information is missing and we can't continue
- `status: wont do/fix`
    - This will not be worked on
- `status: blocked`
    - This is blocked by another problem and can't be worked on
    - Variants:
        - `status: blocked-by-upstream`
        - `status: blocked-by-backend`
        - `status: blocked-by-frontend`
- `status: abandoned`
    - Nobody is working on this anymore

## Other

- `good first issue`
    - Good for newcomers

Of course, each project in our org would be free to add more labels, if they help to structure their work. Added labels should adhere to the structure laid out here, e.g. they should use `topic: detail` to group into categories when sorted alphabetically.

Please comment below if you think I missed something important or have other questions.

**Comment by InnayTool** [comment]
Is _epic_ the label name or should it be replaced by a topic?
Why do we need feature and feature request?

**Comment by DerMolly** [comment]
feature for PRs and feature request for Issues?

**Comment by InnayTool** [comment]
yes but why should we separate that? It is separated by issues and PRs.

**Comment by SISheogorath** [comment]
I mean, I don't want to spoil the fun here, but do we really need that level of organisation? I feel like in the bootstrapping process of a software company. Yes, 2.0 needs quite some work, but I don't know if that level of organisation will make things easier.

I suspect for example that the next step is to ensure that all issues and PRs are properly labeled. Soon requiring a bot to do so.

Personally I consider repositories where issues and PRs are organized that detailed rather unwelcoming. Starting with the fact that someone else decides whether or not the priority of my problem is "low", "medium" or "high". Yes, we already do that to people in form of an answer, but that, from my perspective, way more welcoming that just having a label that states such a thing.

It all just feels a lot more like factory than cosy place to hang out.

**Comment by DerMolly** [comment]
I don't think improving the handling of issues and PRs is a bad thing. The project had many things laying abandoned in the past and that's not really great for anyone. Maybe this is too much, but I rather try to improve and overshoot.

**Comment by mrdrogdrog** [comment]
Nobody says, that we will stop telling the people why an issue is important or not. But these labels makes it a lot easier to filter, search and look through. They are just an addition.

**Comment by davidmehren** [comment]
I don't follow your notion of "organization implies factory".

While working on 2.0 and the 1.7 releases, I was often confused by our current labeling system, which seemed applied inconsistently and made it harder e.g. to find out what still needs to be worked on. My proposal intends to make the labels clearer and easier to use, for us maintainers as well as for all other contributors.

**Comment by DerMolly** [comment]
@InnayTool I'd think we should just have 'type: feature' then.

**Comment by SISheogorath** [comment]
I mean, I'm saying by no means that the current labelling system is good. It's chaotic, it's quite unstructured and it's simply historically grown. So improvements are more than welcome. But looking at the system above, I feel JIRAed and that's definitely something I try to avoid as good as I can.

I also don't think that "organization implies factory", but usually "epic implies factory". epic are usually topics that are organized to push a feature or similar forward and to coordinate work on it. Means it indicates a feature-driven workflow. And that's what I associate with a factory or company. In contrast to the current, totally granted, less effective and more chaotic people-driven workflow.

It might only be in my mind, but feature-driven workflows try to optimize productivity, working out supply chains and create a factory-like setup, where there are stations setup in an order to produce a feature. A people-driven workflow on the other hand is more like this place everyone visits now and then, there are a bunch of workbenches and tools and supplies and everyone can fiddles their stuff together, and once it's done, they show it to other people in the room and everyone enjoys the work.

Maybe that makes things clear?

**Comment by davidmehren** [comment]
After discussing with @InnayTool and @DerMolly I removed the "epic" label, as the use-case seems to be covered by the "discussion" label.

**Comment by davidmehren** [comment]
I now also merged `feature` and `feature request`.

If there are no objections, I will go ahead and implement the labels tomorrow evening.

**Comment by mrdrogdrog** [comment]
This can be closed because its already done, right?

**Comment by mrdrogdrog** [comment]
Did you update renovate so it uses the correct label?

**Comment by DerMolly** [comment]
this seems finished and can be closed?
